### ISSUE TITLE
[XLA:CPU] oneDNN Softmax gives inaccurate results

### ISSUE THREAD

**Issue by penpornk** [openxla/xla#11772]
From https://github.com/google/jax/issues/20856,
`jax.nn.softmax` returns different results under `jax.jit` because the `jax.jit` version calls oneDNN custom call.

Code:
```
import jax
import jax.numpy as jnp

x = jax.random.normal(jax.random.key(0), (3, 1, 1))

def f(x):
    return jax.nn.softmax(x, axis=0)

print(f(x))
print(jax.jit(f)(x))
```

Results
```
[[[0.75250584]]

 [[0.0755428 ]]

 [[0.17195134]]]
[[[1.]]

 [[1.]]

 [[1.]]]
```

Need to investigate why the oneDNN softmax custom call version is giving all 1s.

**Comment by jakevdp** [comment]
Please also note that this behavior is not observed at HEAD, because the HLO generated by `softmax` changed after https://github.com/google/jax/pull/20643 and so the rewrite rule no longer emits the custom call.

**Comment by penumallas1261** [comment]
Hi @penpornk ,
I can see the latest code is fixed the issue and following is the [gist](https://colab.sandbox.google.com/gist/penumallas1261/fc148a51768fc8065cbb9fd052b829f8/untitled0.ipynb), Please feel free to close the issue if it's resolved.

**Comment by penpornk** [comment]
Hi @penumallas1261,
Thank you for bringing this up! I believe the issue may have been hidden as oneDNN is temporarily disabled on JAX and in XLA. But it will take a while before we re-enabled it. So I think we can close this issue for now. We can reopen it later if/when we use oneDNN primitives by default again.

**Comment by penumallas1261** [comment]
Internal reproducer submitted in cl/781834261.
